### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by amita-roy**
Created game, board and player classes
Updated the classes with their individual functionality
Removed the comments
Added game logic 
Added restart functionality
Fixed rubocop error
Fixed linter error

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ Metrics/ClassLength:
   Max: 150
 Metrics/BlockLength:
   ExcludedMethods: ['describe']
-  Max: 30
+  Max: 55
```
Comment (krys2fa) [inline .rubocop.yml]: - [x] [MANDATORY] Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. 
Instead, you can disable the cops within your source code as seen [here](https://rubydoc.info/gems/rubocop/0.24.1/frames#disabling-cops-within-source-code).

File: bin/main.rb [hunk 2]
```diff
@@ -1,108 +1,111 @@
 #!/usr/bin/env ruby
 
-# Creat the Game, Board and Player classes
-# Create the Game and Board objects
+require './lib/player'
+require './lib/game'
+require './lib/board'
```
Comment (krys2fa) [inline bin/main.rb]: - [x] [OPTIONAL] By convention, the use of `require_relative` is preferred to `require`.
`require_relative` is a subset of `require` and is more efficient in referencing files. Read more [here](https://medium.com/@ellishim/understanding-require-vs-require-relative-vs-require-all-80e3b26d89e6).

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require './lib/utils'
+
+class Board
+  include Utils
+
+  attr_accessor :display_board
```
Comment (krys2fa) [inline lib/board.rb]: - [x] [OPTIONAL] Only use `attr_accessors` if you really need to. Using `attire_accessors` means that the property can be set to anything from outside the class. ( [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/)).

**Review by krys2fa — CHANGES_REQUESTED** [review]
Hello @cristianCeamatu and @Amita-Roy,
## This project is awesome!   🔥 🔥 🔥 
- It has a creative welcome message making it stand out. ⭐ 
- It is really interactive making you forget that you are in a terminal/command line. ⭐ 
- It has clean and good organization of code. ⭐ 
- It displays good implementation of OOP. ⭐ 
 
### Status: Changes Required ♻️ 
Firstly, the README file contains images/gifs that are not showing up. Also, the link to the live demo seems to be broken as well.

See below for some more observations to make your project even better (if that's even possible 😄).

Request for another review afterwards to get this project approved.

Cheers and Happy Coding 🖥️  😃 

###### You can contact me on slack (Chris Amanor) for any clarification.

File: .rubocop.yml (hunk 1 above)
Comment (amita-roy) [reply .rubocop.yml]: update it back to 30 as adviced.

File: bin/main.rb (hunk 2 above)
Comment (amita-roy) [reply bin/main.rb]: Update require_relative as adviced

File: lib/board.rb (hunk 3 above)
Comment (amita-roy) [reply lib/board.rb]: Update attr_accessor to attr_reader in board class

**Comment by Dandush03** [comment]
# Status: :rocket: :star: Approved! :star: :rocket: 

## Keep Up the good work!

# Happy Coding :nerd_face: :rocket:

**Pull request merged**